tests/lib/prepare.sh

```diff
@@ -881,6 +881,33 @@ EOF
         exit 1
     fi
 
+    
+    
+    
+    # download the core20 snap manually from the specified channel for UC20
+    if os.query is-core20; then
+        snap download core20 --channel="$BASE_CHANNEL" --basename=core20
+
+        
+        # we want to download the specific channel referenced by $BASE_CHANNEL, but
+        # if we just seed that revision and $BASE_CHANNEL != $IMAGE_CHANNEL, then 
+        # immediately on booting, snapd will refresh from the revision that is 
+        # seeded via $BASE_CHANNEL to the revision that is in $IMAGE_CHANNEL, so to
+        # prevent that from happening (since that automatic refresh will confuse
+        # spread and make tests fail in awkward, confusing ways), we unpack the snap
+        # and re-pack it so that it is not asserted and thus won't be automatically
+        # refreshed
+        if [ "$IMAGE_CHANNEL" != "$BASE_CHANNEL" ]; then
+            sudo unsquashfs -d core20-snap core20.snap
+            sudo snap pack --filename=core20-repacked.snap core20-snap
+            mv core20-repacked.snap $IMAGE_HOME/core20.snap
```

Comment:
This is handled by the restore helper (as I've been told by Sergio :-) )

Reply:
I'm just wondering whether there's any risk that the test will run out of disk space, if this is done in the prepare step. But if there isn't, fine with me :-)

Reply:
I think the core20 snap is small enough that it's not a significant issue, but if we do run into space issues perhaps we should re-evaluate this

Reply:
so, in this case the restore does not clean up the files because the cleanup is done in the restore-each which runs in the test path for each test, in this case the path is different and the restore code for the suite does not clean it.

Reply:
after that code we have a line which cleans the snaps downloaded 
# rm -f ./pc-kernel_*.{snap,assert} ./pc_*.{snap,assert} ./snapd_*.{snap,assert}

We should inclued there the cleanup for core20 snap as well

Reply:
thanks for clarifying @sergiocazzolato I'll add this

Reply:
@sergiocazzolato what about deleting the files in $IMAGE_HOME, afaict we aren't deleting those files too, just the ones in the current working directory when this runs?

Reply:
the files in $IMAGE_HOME are then used to boot the image so if we want to clean that needs to be once the core image boots.  I'd just delete the files in the working directory, then in a following pr we can explore if /home/image needs to be deleted as well